src/EditorFeatures/Text/Extensions.cs

```diff
@@ -37,7 +40,7 @@ public static SourceText AsText(this ITextSnapshot textSnapshot)
             => SnapshotSourceText.From(textSnapshot);
 
         internal static SourceText AsRoslynText(this ITextSnapshot textSnapshot, Encoding encoding)
-            => new SnapshotSourceText.ClosedSnapshotSourceText(textSnapshot, encoding);
+            => new SnapshotSourceText.ClosedSnapshotSourceText(((ITextSnapshot2)textSnapshot).TextImage, encoding);
```

Comment:
http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Workspaces/EditorTextFactoryService.cs,71

looks like still hold true for now. text from open file always comes from editor not from our text factory.